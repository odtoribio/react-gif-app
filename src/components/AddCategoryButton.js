import React, { useState } from 'react'
import PropTypes from 'prop-types'  

export const AddCategoryButton = ({ addCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value)
  }

  const handleOnSubmit = (e) => {
    e.preventDefault()

    if (inputValue){
      addCategory(inputValue)
      setInputValue('')
    }
  }
  
  return (
    <form onSubmit={ handleOnSubmit }>
      <input 
        type="text"
        value={ inputValue }
        onChange= { handleInputValue }
      />
    </form>
  )
}

AddCategoryButton.propTypes = {
  addCategory: PropTypes.func.isRequired,
}
