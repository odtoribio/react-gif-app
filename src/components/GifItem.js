import React from 'react'
import PropTypes from 'prop-types';

export const GifItem = (props) => {
  const { title, url } = props;
  return (
    <div className="card animate__animated animate__zoomIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
