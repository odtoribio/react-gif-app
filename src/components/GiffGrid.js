import React from 'react';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GiffGrid = ( { category } ) => {

  const { data:gifs, loading } = useFetchGifs(category)

  return (
    <>
      <h3>{ category }</h3>
      { loading && 'Loading...' }
      <div className="card-grid">
          {
            gifs.map( gif => (
              <GifItem 
                key={gif.id} 
                gif={ gif }
              />
            ))
          }
      </div>
    </>
  )
}
