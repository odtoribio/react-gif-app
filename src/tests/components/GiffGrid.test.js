import { shallow } from 'enzyme'
import { GiffGrid } from '../../components/GiffGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe( 'Test GiffGrid component', () => {

  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('should render the component', () => {

    useFetchGifs.mockReturnValue({
      data:[],
      loading: true
    });

    const GiffGridComponent = shallow(<GiffGrid category="One Punch" />);

    expect(GiffGridComponent).toMatchSnapshot()
  })
  
  test('should show items when images load with useFetchGif', () => {

    const gifs = [{
      id:"ABC",
      url:"http://test.com/image.jpg",
      title:"title"
    }]
    
    useFetchGifs.mockReturnValue({
      data:gifs,
      loading: false
    });

    const GiffGridComponent = shallow(<GiffGrid category="One Punch" />);
    expect(GiffGridComponent.find('p').exists()).toBe(false)
    expect(GiffGridComponent.find('GifItem').length).toBe(gifs.length)
  })
  
})
