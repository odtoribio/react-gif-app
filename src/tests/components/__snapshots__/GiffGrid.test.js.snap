// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GiffGrid component should render the component 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  Loading...
  <div
    className="card-grid"
  />
</Fragment>
`;
